### PR TITLE
[Price Impact] - Fix price calculation and price warnings

### PR DESCRIPTION
**Pull request by W3stside**
Fixes price calculation and fixes warnings from small calc errors

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
   * **🎩 [Dapp](https://pr1580--dexreact.review.gnosisdev.com)**: Testing web app
   * **📚 [Storybook](https://pr1580--dexreact.review.gnosisdev.com/storybook/)**: Component stories

File: src/components/TradeWidget/index.tsx [hunk 1]
```diff
@@ -301,10 +301,10 @@ const TradeWidget: React.FC<TradeWidgetProps> = ({
   useEffect(() => {
     if (priceValue && sellValue) {
       // If price is quoted in sell tokens, we use it, otherwise we use the inverse
-      const priceUsedForReceiveAmount = sellToken === quoteToken ? priceValue : priceInverseValue
+      const priceUsedForReceiveAmount = sellToken.address === quoteToken.address ? priceValue : priceInverseValue
```
Comment (anxolin) [inline src/components/TradeWidget/index.tsx:304]: is this related to the issue? was the problem related to "identity"?

File: src/components/TradeWidget/utils.ts [hunk 2]
```diff
@@ -16,7 +17,11 @@ export function calculateReceiveAmount(priceValue: string, sellValue: string): s
 
     if (sellAmount && price) {
       const receiveBigNumber = sellAmount.dividedBy(price)
-      receiveAmount = receiveBigNumber.isNaN() || !receiveBigNumber.isFinite() ? '0' : receiveBigNumber.toString(10)
+      // Receive at least input value
+      // uses amount floor < 0.0001
+      // but Limit Price set shows full amount
```
Comment (anxolin) [inline src/components/TradeWidget/utils.ts:22]: I get the change below, but I didn't understand the comment

File: src/components/trade/PriceSuggestions/PriceSuggestionItem.tsx [hunk 3]
```diff
@@ -21,54 +21,67 @@ export interface Props {
...
   priceFormatted: string
-  invertedPriceFormatted: string
+  inversePriceFormatted: string
+  inversePriceRaw: string
+  priceRaw: string
 }
 
-function formatPriceToPrecision(price: BigNumber): string {
-  return price.toFixed(PRICE_ESTIMATION_PRECISION)
+const LOW_PRICE_FLOOR = new BigNumber(0.0001)
```
Comment (anxolin) [inline src/components/trade/PriceSuggestions/PriceSuggestionItem.tsx:29]: I think it's better to use string in the constructor, so there's never float point issues

File: src/components/trade/PriceSuggestions/PriceSuggestionItem.tsx [hunk 4]
```diff
@@ -21,54 +21,67 @@ export interface Props {
 
 interface FormattedPrices {
   priceFormatted: string
-  invertedPriceFormatted: string
+  inversePriceFormatted: string
+  inversePriceRaw: string
+  priceRaw: string
```
Comment (anxolin) [inline src/components/trade/PriceSuggestions/PriceSuggestionItem.tsx:26]: good call

Comment (anxolin) [reply src/components/trade/PriceSuggestions/PriceSuggestionItem.tsx:26]: good call

I would expect raw being a BigNumber (judging by the name)
Since all these are "formatted" (all are strings), I would suggest:
- priceLabel
- inversePriceLabel
- priceValue
- inversePriceValue

I think, then the `<button>` is more clear

**Review by anxolin — APPROVED** [review]

File: src/components/TradeWidget/index.tsx (hunk 1 above)
Comment (W3stside) [reply src/components/TradeWidget/index.tsx:304]: I made this change as I was seeing a lot of re-renders in this hook, not sure if the references between the objects `sellToken` and `quoteToken` is stable. anyways can remove

File: src/components/TradeWidget/utils.ts (hunk 2 above)
Comment (W3stside) [reply src/components/TradeWidget/utils.ts:22]: changed it
